src/odemis/gui/util/img.py

```diff
@@ -1159,12 +1160,13 @@ def bar_plot(ctx, data, data_width, range_x, data_height, range_y, client_size,
     ctx.fill()
 
 
-def spectrum_to_export_data(proj, raw):
+def spectrum_to_export_data(proj, raw, vp=None):
     """
     Creates either raw or WYSIWYG representation for the spectrum data plot
 
     proj (DataProjection): DataProjection of the spectrum to export
     raw (boolean): if True returns raw representation
+    vp (Viewport): Viewport object
```

Comment:
Note: ideally, you wouldn't pass a viewport, but a `gui.model.StreamView`. That (special) view would have the corresponding VAs `hrange` and `vrange`. The NavigablePlotViewport would just read/write them, whenever the user moves or zooms around. Anyway, for now, let's keep it like this.